### PR TITLE
Jha sup ben info page

### PR DESCRIPTION
**Pull request by Ian-c-cameron**
### Please check if the PR fulfills these requirements:

- [X] After these changes, the app was run and still works as expected
- [ ] Unit tests for these changes were added (if applicable)
- [X] All existing unit tests were run and still pass

### Please specify the type of change this PR introduces (Bug fix, feature addition, content update, chore, etc.):
First draft of the Supplemental Benefits info page.

### CODE REVIEW

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]

File: jha/src/components/SuppBenWidget.vue [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  convertNumberToFormattedString,
+  calculateAge,
+} from 'common-lib-vue';
+
+export default {
+  name: "SuppBenWidget",
+  props: {
+    value: {
+      type: Object,
+      default: {},
```
[the remark this answers is not in the document]
Comment (Ian-c-cameron) [reply jha/src/components/SuppBenWidget.vue]: fixed with data handling changes

File: jha/src/components/SuppBenWidget.vue [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+  },
+  computed: {
+    // Income
+    sbTotalHouseholdIncome() {
+      let income = this.stringToFloat(this.value.ahSBIncome);
+      if( this.value.hasSpouse) {
+        income += this.stringToFloat(this.value.spouseSBIncome);
+      }
+      income = (income > 0) ? income : 0;
+      this.value.sbTotalHouseholdIncome = income;
```
[the remark this answers is not in the document]
Comment (Ian-c-cameron) [reply jha/src/components/SuppBenWidget.vue]: fixed with data handling changes

File: jha/src/components/SuppBenWidget.vue [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+      this.value.spouseDeduction = deduction;
+      return deduction;
+    },
+    spouse65Deduction() {
+      let deduction = this.ageOver65(this.value.spouseBirthDate) ? 3000 : 0;
+      this.value.spouse65Deduction = deduction;
+      return deduction;
+    },
+    numChildren() {
+      return !!this.value.children ?  this.value.children.length : 0;
```
[the remark this answers is not in the document]
Comment (Ian-c-cameron) [reply jha/src/components/SuppBenWidget.vue]: fixed with data handling changes

**Review by harrymaynard-maximus — CHANGES_REQUESTED** [review]
Maybe I should have called this out earlier, but I don't think the widget should be doing calculations then sending it back via some event. I think the calculations should be done purely in the SB page component, and passed to the widget for rendering only.

That way you can eliminate the updating of data from the widget component, and pass data to the store directly from the page component.

File: jha/src/data-types/supp-ben-data.js [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+  #hasDisabilityCredit;
+  #selectedDisabilityRecipients;
+  #numDisabilityChildren;
+  #hasRDSP;
+  #rdspAmount;
+  #hasAttendantNursingExpenses;
+  #selectedAttendantNursingRecipients;
+  #numAttendantNursingChildren;
+  #attendantNursingReceipts;
+  #qualificationThreshhold;
```
Comment (harrymaynard-maximus) [inline jha/src/data-types/supp-ben-data.js:23]: What's the purpose of the "#" prefix? I've never seen that before...

Comment (Ian-c-cameron) [reply jha/src/data-types/supp-ben-data.js:23]: While JS doesn't have any provision for having private instance variables(you can't stop someone accessing directly without using getters or setters) it is a standard convention that private variables are prefixed with a '#'.  Many linters will even throw out errors if someone tries to access one from outside the class.

Comment (harrymaynard-maximus) [reply jha/src/data-types/supp-ben-data.js:23]: Interesting. Appears to be a new JS feature. I'm fine with this, but please make sure it still works in IE11. Babel will likely transpile this, but just to be sure.

**Review by harrymaynard-maximus — APPROVED** [review]
Looks good.

File: jha/src/data-types/supp-ben-data.js (hunk 4 above)
Comment (Ian-c-cameron) [reply jha/src/data-types/supp-ben-data.js:23]: I have tested in IE and it seems to work fine... as this is only a naming convention it shouldn't make a difference to behavior even if babel doesn't transpile it.

**Pull request merged**